src/engine.rs

```diff
@@ -434,30 +428,30 @@ impl Reedline {
 
             // Move right from this position to the column we were at
             while &buffer[position..(position + 1)] != "\n" && num_of_move_lefts > 0 {
-                self.line_buffer.move_right();
-                position = self.line_buffer.offset();
+                self.editor.move_right();
+                position = self.editor.offset();
                 num_of_move_lefts -= 1;
             }
         }
     }
 
     fn down_command(&mut self) {
         // If we're at the top, then:
-        if !self.line_buffer.get_buffer()[self.line_buffer.offset()..].contains('\n') {
+        if !self.editor.get_buffer()[self.editor.offset()..].contains('\n') {
             // If we're at the top, move to previous history
             self.next_history();
         } else {
             // If we're not at the top, move up a line in the multiline buffer
-            let mut position = self.line_buffer.offset();
+            let mut position = self.editor.offset();
             let mut num_of_move_lefts = 0;
-            let buffer = self.line_buffer.get_buffer().to_string();
+            let buffer = self.editor.get_buffer().to_string();
 
             // Move left until we're looking at the newline
             // Observe what column we were on
             while position > 0 && &buffer[(position - 1)..position] != "\n" {
-                self.line_buffer.move_left();
+                self.editor.move_left();
                 num_of_move_lefts += 1;
-                position = self.line_buffer.offset();
+                position = self.editor.offset();
             }
```

Comment:
I intend to pull some cohesive pieces out of this method into the line_buffer and the editor. Then use them here. Hopefully, we will be able to simplify this logic.